ambry-messageformat/src/test/java/com.github.ambry.messageformat/MessageSievingInputStreamTest.java

```diff
@@ -560,109 +653,348 @@ private void testDeletedBlobs(short blobVersion, BlobType blobType) throws IOExc
         msgInfoList.add(msgInfo2);
         msgInfoList.add(msgInfo3);
 
-        MessageSievingInputStream validMessageDetectionInputStream =
-            new MessageSievingInputStream(inputStream, msgInfoList, new MockIdFactory(), new MetricRegistry());
-        Assert.fail("IllegalStateException should have been thrown due to delete record ");
+        new MessageSievingInputStream(inputStream, msgInfoList, transformers, new MetricRegistry());
+        if (!options.isEmpty()) {
+          Assert.fail("IOException should have been thrown due to delete record ");
+        }
+      }
+    } catch (IOException e) {
+      if (options.isEmpty()) {
+        Assert.fail("No exceptions should have occurred");
       }
-    } catch (IllegalStateException e) {
-      Assert.assertTrue("IllegalStateException thrown as expected ", true);
     }
     headerVersionToUse = Message_Header_Version_V1;
   }
 
-  private boolean verifyBlob(MessageSievingInputStream validMessageDetectionInputStream, short headerVersion,
-      int headerSize, int blobEncryptionRecordSize, int blobPropertiesRecordSize, int userMetadataSize, int blobSize,
-      StoreKey key, int blobPropertiesSize, String serviceId, short accountId, short containerId, byte[] encryptionKey,
-      byte[] usermetadata, byte[] data, short blobVersion, BlobType blobType) throws IOException {
-    // verify header
-    byte[] headerOutput = new byte[headerSize];
-    validMessageDetectionInputStream.read(headerOutput);
-    ByteBuffer headerBuf = ByteBuffer.wrap(headerOutput);
-    Assert.assertEquals(headerVersion, headerBuf.getShort());
-    Assert.assertEquals(blobEncryptionRecordSize + blobPropertiesRecordSize + userMetadataSize + blobSize,
-        headerBuf.getLong());
-    if (headerVersion == Message_Header_Version_V2) {
-      if (blobEncryptionRecordSize == 0) {
-        Assert.assertEquals(Message_Header_Invalid_Relative_Offset, headerBuf.getInt());
-      } else {
-        Assert.assertEquals(headerSize + key.sizeInBytes(), headerBuf.getInt());
-      }
+  /**
+   * Test the case where there are deprecated messages.
+   * @throws Exception
+   */
+  @Test
+  public void testDeprecatedMsgTransformation() throws Exception {
+    testDeprecatedMsg(Blob_Version_V1, BlobType.DataBlob, MessageFormatRecord.Message_Header_Version_V1);
+    testDeprecatedMsg(Blob_Version_V2, BlobType.DataBlob, MessageFormatRecord.Message_Header_Version_V2);
+    testDeprecatedMsg(Blob_Version_V2, BlobType.MetadataBlob, MessageFormatRecord.Message_Header_Version_V2);
+  }
+
+  private void testDeprecatedMsg(short blobVersion, BlobType blobType, short headerVersionToUse) throws Exception {
+    MessageFormatRecord.headerVersionToUse = headerVersionToUse;
+
+    // MessageSievingInputStream contains put records for 2 valid blobs and 1 corrupt blob
```

Comment:
is this comment accurate? (any corruption?)

Reply:
fixed - thanks.